### ISSUE TITLE
offline api

### ISSUE THREAD

**Issue by criloz** [nebtex/taurus#2]
@Ignusmart 

for now implement the offline api with a class

```
 class Offline
```

it has a method called query that should be similar to this

offline = Offline (hashmap)
query(app:string, pattern:string, function(  [elements], [newelement], prevResult) result{})

the query method should return an observable,

[elements] is the list of element that match the pattern in the app, 
when new elements are stored in the hahmap [newelement] should contain them, 

prevResult is the previus result of the query

result is the result of the function wraped in a observable. 

there is  a hashmap per app

**Comment by Ignusmart** [comment]
@criloz can you please answer to me this question, is the ` function( [elements], [newelement], prevResult) ` a callback ? I mean, that function is a param of the query function?

**Comment by criloz** [comment]
@Ignusmart yes is a callback each time that a new element is add to the hashmap that function should be called.

**Comment by criloz** [comment]
@Ignusmart also you will need to finish this task first to know if a new insertion match the x pattern 

https://github.com/nebtex/hashmaps/issues/1

you should do what it says

**Comment by criloz** [comment]
@Ignusmart but first see waht it receive in the change object please check the mobx documentation for that, 

you can use autorun, or add an observer to the hasmap add or delete x, key and see how is being called the callback that you put in the observe function try to see if you can use those values in mini match for match the pattern.,
